### DISCUSSION TITLE
error using CircuitToEinsum `NotImplementedError: circuit from qiskit not supported`

### DISCUSSION THREAD

**Discussion by d-bharadwaj** [NVIDIA/cuQuantum#161] [Q&A] [answered]
Hi team! My cuquantum cannot infer the package that defines the circuit object from qiskit. I am on WSL using qiskit version 0.43.2 and quantum-python version 24.3.0. I encounter this issue in a different project but here is some short code that replicates the error.

Has anyone encountered this before?

```python

from cuquantum import cutensornet as cutn
from cuquantum import contract, CircuitToEinsum
import cupy as cp 
from qiskit.circuit import QuantumCircuit, Parameter

num_qubits = 1
measurement_ops = 'Z'
light_cone  = True 

# Define the quantum circuit with one qubit and two parameters
theta = Parameter('θ')
phi = Parameter('φ')

parameters = [0.5,0.5]

circuit = QuantumCircuit(num_qubits)
circuit.rx(theta, 0)
circuit.ry(phi, 0)

circuit = circuit.bind_parameters({theta: parameters[0], phi: parameters[1]})
converter = CircuitToEinsum(circuit, backend='cupy')  
einsum_expression, tensor_operands = converter.expectation(measurement_ops, light_cone)
``` 

and this is the error : 
` File "/home/<username>/miniconda3/envs/cafqa/lib/python3.10/site-packages/cuquantum/cutensornet/_internal/circuit_converter_utils.py", line 89, in infer_parser 
raise NotImplementedError(f'circuit from {base} not supported')
NotImplementedError: circuit from qiskit not supported`

**Comment by yangcal** [comment]
Hello Dhanvi,

Thanks for reporting. This is a miss when we bumped the support of qiskit>=1.0. 

For a temporary fix, you may:

1. If you're comfortable editing the source file, change `ModuleNotFoundError` to `ImportError` https://github.com/NVIDIA/cuQuantum/blob/8d3bbce93c64789f61898f033cae49f83329f933/python/cuquantum/cutensornet/_internal/circuit_parser_utils_qiskit.py#L12
2. Updating qiskit to >=1.0, this may affect your current qiskit scripts though but maybe better for your project in the long term to adhere to latest qiskit.

We will fix this issue in the next cuquantum-python release.

Thanks

**Comment by yangcal** [comment, marked as answer]
This issue has been resolved in 24.11 release [here](https://github.com/NVIDIA/cuQuantum/blob/v24.11.0/python/cuquantum/cutensornet/_internal/circuit_parser_utils_qiskit.py). 

Closing the discussion

**Reply by d-bharadwaj** [reply]
Thanks for letting me know! @yangcal
